src/Kiota.Builder/Writers/Php/CodeMethodWriter.cs

```diff
@@ -71,21 +71,21 @@ public override void  WriteCodeElement(CodeMethod codeElement, LanguageWriter wr
             if(inherits)
                 writer.WriteLine("parent::__construct();");
             foreach(var propWithDefault in parentClass.GetPropertiesOfKind(
-                                                                            CodePropertyKind.BackingStore,
-                                                                            CodePropertyKind.RequestBuilder,
-                                                                            CodePropertyKind.UrlTemplate,
-                                                                            CodePropertyKind.PathParameters)
-                                            .Where(x => !string.IsNullOrEmpty(x.DefaultValue))
-                                            .OrderByDescending(x => x.Kind)
-                                            .ThenBy(x => x.Name))
+                    CodePropertyKind.BackingStore,
+                    CodePropertyKind.RequestBuilder,
+                    CodePropertyKind.UrlTemplate,
+                    CodePropertyKind.PathParameters)
+                .Where(x => !string.IsNullOrEmpty(x.DefaultValue))
+                .OrderByDescending(x => x.Kind)
+                .ThenBy(x => x.Name))
             {
                 var isPathSegment = propWithDefault.IsOfKind(CodePropertyKind.PathParameters);
-                writer.WriteLine($"$this->{propWithDefault.NamePrefix}{propWithDefault.Name.ToFirstCharacterLowerCase()} = {(isPathSegment ? "[]" :propWithDefault.DefaultValue.ReplaceDoubleQuoteWithSingleQuote())};");
+                writer.WriteLine($"$this->{propWithDefault.Name.ToFirstCharacterLowerCase()} = {(isPathSegment ? "[]" :propWithDefault.DefaultValue.ReplaceDoubleQuoteWithSingleQuote())};");
             }
-            foreach(var propWithDefault in parentClass.GetPropertiesOfKind(CodePropertyKind.AdditionalData) //additional data and backing Store rely on accessors
+            foreach(var propWithDefault in parentClass.GetPropertiesOfKind(CodePropertyKind.AdditionalData, CodePropertyKind.Custom) //additional data and custom properties rely on accessors
                 .Where(x => !string.IsNullOrEmpty(x.DefaultValue))
                 .OrderBy(x => x.Name)) {
-                writer.WriteLine($"$this->{propWithDefault.Name.ToFirstCharacterLowerCase()} = {propWithDefault.DefaultValue};");
+                writer.WriteLine($"$this->set{propWithDefault.SymbolName.ToFirstCharacterUpperCase()}({propWithDefault.DefaultValue.ReplaceDoubleQuoteWithSingleQuote()});");
```

Comment:
you might get into a situation where your accessor name between multiple inherited types doesn't match. Can you rebase onto #1739 and look at the changes I've made in the Go CodeMethod writer for the default values please?

Reply:
Rebased and updated.